src/test/java/dev/blachut/svelte/lang/codeInsight/SvelteHighlightingTest.kt

```diff
@@ -0,0 +1,326 @@
+        functionSignaturesInspection.myCheckGuessedTypes = true
+        l.add(functionSignaturesInspection)
+        l.add(JSValidateJSDocInspection())
+        l.add(JSUndeclaredVariableInspection())
+        l.add(XmlDuplicatedIdInspection())
+        l.add(ES6CheckImportInspection())
+        l.add(XmlInvalidIdInspection())
+        l.add(HtmlUnknownTagInspection())
+        l.add(HtmlUnknownBooleanAttributeInspection())
+        l.add(HtmlUnknownAttributeInspection())
+        l.add(XmlUnboundNsPrefixInspection())
+        l.add(JSUnusedLocalSymbolsInspection())
+        l.add(JSPotentiallyInvalidConstructorUsageInspection())
+        l.add(JSUnnecessarySemicolonInspection())
+        l.add(JSLastCommaInArrayLiteralInspection())
+        l.add(JSLastCommaInObjectLiteralInspection())
+        l.add(JSReferencingMutableVariableFromClosureInspection())
+        l.add(JSPotentiallyInvalidUsageOfThisInspection())
+        l.add(JSPotentiallyInvalidUsageOfClassThisInspection())
+        l.add(JSPotentiallyInvalidTargetOfIndexedPropertyAccess())
+        l.add(JSUndefinedPropertyAssignmentInspection())
+        l.add(JSClosureCompilerSyntaxInspection())
+        l.add(JSCommentMatchesSignatureInspection())
+        l.add(JSFileReferencesInspection())
+        l.add(JSUnusedGlobalSymbolsInspection())
+        l.add(JSReferencingArgumentsOutsideOfFunctionInspection())
+        l.add(JSAnnotatorInspection())
+        l.add(JSUnusedAssignmentInspection())
+        l.add(UnreachableCodeJSInspection())
+        l.add(TypeScriptValidateTypesInspection())
+        l.add(TypeScriptValidateJSTypesInspection())
+        l.add(TypeScriptUnresolvedFunctionInspection())
+        l.add(TypeScriptUnresolvedVariableInspection())
+        l.add(TypeScriptAccessibilityCheckInspection())
+        l.add(TypeScriptCheckImportInspection())
+        l.add(TypeScriptDuplicateUnionOrIntersectionTypeInspection())
+        l.add(PointlessBooleanExpressionJSInspection())
+        l.add(TypeScriptValidateGenericTypesInspection())
+        l.add(TypeScriptRedundantGenericTypeInspection())
+
+        //CF inspections
+        l.add(JSSuspiciousTypeGuardInspection())
+        l.add(JSObjectNullOrUndefinedInspection())
+        l.add(SvelteUnresolvedComponentInspection())
+        return l
+    }
+
+    override fun setUp() {
+        super.setUp()
+        myFixture.enableInspections(*configureDefaultLocalInspectionTools().toTypedArray())
+    }
+
+    private fun configureComponentWithAttribute(name: String) {
+        myFixture.configureByText(name, "<script>export var test=\"\"</script>")
+    }
+
+    fun testUnresolvedComponent() {
+        configureComponentWithAttribute("Foo.svelte")
+        myFixture.configureByText("Usage.svelte", "<<error>Fo<caret>o</error> />")
+        myFixture.testHighlighting()
+        myFixture.launchAction(myFixture.findSingleIntention("Insert 'import Foo from \"./Foo.svelte\"'"))
```

Comment:
should intentions be tested in the same class?

Reply:
actually it is ok to test a quick fix in highlighting tests if the fix is provided by highlighting information because we have to test these things together anyway

Reply:
makes sense